components/admin_console/permission_schemes_settings/permissions_tree/permissions_tree.jsx

```diff
@@ -0,0 +1,187 @@
+            integrationsGroup.permissions.push('manage_webhooks');
+        }
+        if (config.EnableOAuthServiceProvider === 'true' && integrationsGroup.permissions.indexOf('manage_oauth') === -1) {
+            integrationsGroup.permissions.push('manage_oauth');
+        }
+        if (config.EnableCommands === 'true' && integrationsGroup.permissions.indexOf('manage_slash_commands') === -1) {
+            integrationsGroup.permissions.push('manage_slash_commands');
+        }
+        if (config.EnableCustomEmoji === 'true' && integrationsGroup.permissions.indexOf('manage_emojis') === -1) {
+            integrationsGroup.permissions.push('manage_emojis');
+        }
+        if (scope === 'team_scope' && this.groups[0].id !== 'teams_team_scope') {
+            this.groups[0].id = 'teams_team_scope';
+        }
+    }
+
+    onClickEditPostTimeLimitButton = () => {
+        this.setState({editTimeLimitModalIsVisible: true});
+    }
+
+    componentDidUpdate(prevProps) {
+        if (this.props.config !== prevProps.config) {
+            this.updateGroups();
+        }
+    }
+
+    toggleGroup = (ids) => {
+        if (this.props.readOnly) {
+            return;
+        }
+        this.props.onToggle(this.props.role.name, ids);
+    }
+
+    render = () => {
+        return (
+            <div className='permissions-tree'>
+                <div className='permissions-tree--header'>
+                    <div className='permission-name'>
+                        <FormattedMessage
+                            id='admin.permissions.permissionsTree.permission'
+                            defaultMessage='Permission'
+                        />
+                    </div>
+                    <div className='permission-description'>
+                        <FormattedMessage
+                            id='admin.permissions.permissionsTree.description'
+                            defaultMessage='Description'
+                        />
+                    </div>
+                </div>
+                <div className='permissions-tree--body'>
+                    <PermissionGroup
+                        key='all'
+                        id='all'
+                        uniqId={this.props.role.name}
+                        selected={this.props.selected}
+                        selectRow={this.props.selectRow}
+                        readOnly={this.props.readOnly}
+                        permissions={this.groups}
+                        additionalValues={this.ADDITIONAL_VALUES}
+                        role={this.props.role}
+                        parentRole={this.props.parentRole}
+                        scope={this.props.scope}
+                        combined={false}
+                        onChange={this.toggleGroup}
+                        root={true}
+                    />
+                </div>
+                <EditPostTimeLimitModal
+                    onClose={() => this.setState({editTimeLimitModalIsVisible: false})}
```

Comment:
Similarly, this should be made into a method of the PermissionsTree or else any time this re-renders, the EditPostTimeLimitModal will as well

Reply:
It wasn't part of my change, but I can move that into a method.

Reply:
Done